plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -627,20 +627,44 @@ type ServiceAffinity struct {
 	labels        []string
 }
 
-func NewServiceAffinityPredicate(podLister algorithm.PodLister, serviceLister algorithm.ServiceLister, nodeInfo NodeInfo, labels []string) algorithm.FitPredicate {
+// serviceAffinityPrecomputation should be run once by the scheduler before looping through the Predicate.  It is a helper function that
+// only should be referenced by NewServiceAffinityPredicate.
+func (s *ServiceAffinity) serviceAffinityPrecomputation() PredicateMetadataModifier {
+	return func(pm *predicateMetadata) {
```

Comment:
Yes but, (by definition) the precomputation takes a predicateMeta and mutates it...